keras_cv/models/segmentation/deeplab/deeplab_v3.py

```diff
@@ -0,0 +1,186 @@
+            Note that this doesn't include the batch dimension. A sample use case
+            is like [512, 512, 3].
+        backbone: a backbone network for the model. Can be a `tf.keras.Model`
+            instance. The supported pre-defined backbone models are:
+            1. "resnet_50", a ResNet50 model
+            2. "dilated_resnet_101", a ResNet 101 model with dilated convolution.
+            3. "mobilenet_v2", a MobileNet V2 model.
+        backbone_weights: pre-trained weights for the backbone model. The weights
+            can be a list of tensors, which will be set to backbone model
+            via `model.set_weights()`. A string file path is also supported, to
+            read the weights from a folder of tensorflow checkpoints. Note that
+            the weights and backbone network should be compatible between each
+            other.
+        decoder: a decode network for segmentation model, eg FPN, or ASPP. The
+            supported premade decodes are: "fpn" or "aspp". The decoder is called on
+            the output of the backbone network to up-sample the feature output.
+        segmentation_head: a optional `tf.keras.Layer` that predict the segmentation
+            mask based on feature from backbone and feature from decoder.
+    """
+
+    def __init__(
+        self,
+        classes,
+        input_size,
+        backbone,
+        backbone_weights,
+        decoder,
+        segmentation_head=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        _assert_tensorflow_model_installed()
+
+        # Common model configs that are shared by all parts of the network
+        # TODO(scottzhu): Expose these params to signature if needed.
+
+        # Common normalization/regularization configs that are shared by all parts of
+        # the network. The following config was for resnet_50.
+        activation = "swish"
+        use_sync_bn = True
+        norm_momentum = 0.99
+        norm_epsilon = 0.001
+        kernel_regularizer = tf.keras.regularizers.l2(0.00002)
+
+        if isinstance(backbone, str):
+            supported_premade_backbone = [
+                "resnet_50",
+                "dilated_resnet_101",
+                "mobilenet_v2",
+            ]
+            if backbone not in supported_premade_backbone:
+                raise ValueError(
+                    "Supported premade backbones are: "
+                    f'{supported_premade_backbone}, received "{backbone}"'
+                )
+            input_spec = tf.keras.layers.InputSpec(shape=[None] + list(input_size))
+            if backbone == "resnet_50":
+                backbone = tfm.vision.backbones.ResNet(
+                    model_id=50,
+                    input_specs=input_spec,
+                    se_ratio=0.0,  # From the default Resnet config
+                    activation=activation,
+                    use_sync_bn=use_sync_bn,
```

Comment:
Do we intend to integrate this with keras CV pretrained weights?

Reply:
yes we should, once we have a set of backbone that is ready to use in kcv, then we should probably replace the one from MG with them.